mantle/cmd/kola/kola.go

```diff
@@ -636,20 +636,29 @@ func getParentFcosBuildBase(stream string) (string, error) {
 		if err != nil {
 			return "", err
 		}
-
+		//as we build for multi-architectures now and inorder to allow failures to be non-fatal, enable them to promote to non-prod streams when any specific previous release is unavailable for that arch.
+		//in that case, releases are searched in reverse order and the most recent release which has the secondary arch is considered.
 		n := len(index.Releases)
-		if n == 0 {
-			// hmmm, no builds; likely a new stream. let's just fallback on testing-devel.
-			msg := fmt.Sprintf("no parent version in build metadata, and no build on stream %s", stream)
-			if stream == "testing-devel" {
-				return "", errors.New(msg)
-			}
-			plog.Infof("%s; falling back to testing-devel", msg)
-			return getParentFcosBuildBase("testing-devel")
+		for i := range index.Releases {
+			//to iterate through the list in reverse order
+  		release := index.Releases[(n - 1 - i)]
+    	for _, commit := range release.Commits {
+  			if commit.Architecture == (kola.Options.CosaBuildArch) {
+  				parentVersion = release.Version
+    			break
+        }
+      }
+    }
+	if parentVersion == "" {
+		//likely no parent version is available as its a new stream; so, fall back to testing-devel
```

Comment:
we can probably do away with this comment since the log messages in the code make it clear what's going on.